sensing/autoware_pointcloud_preprocessor/include/autoware/pointcloud_preprocessor/concatenate_data/concatenate_and_time_sync_node.hpp

```diff
@@ -82,115 +85,87 @@
-
-  double timeout_sec_ = 0.1;
-
-  bool publish_synchronized_pointcloud_;
-  bool keep_input_frame_in_synchronized_pointcloud_;
-  std::string synchronized_pointcloud_postfix_;
-
-  std::set<std::string> not_subscribed_topic_names_;
-
-  /** \brief A vector of subscriber. */
-  std::vector<rclcpp::Subscription<PointCloud2>::SharedPtr> filters_;
-
-  rclcpp::Subscription<geometry_msgs::msg::TwistWithCovarianceStamped>::SharedPtr sub_twist_;
-  rclcpp::Subscription<nav_msgs::msg::Odometry>::SharedPtr sub_odom_;
-
-  rclcpp::TimerBase::SharedPtr timer_;
-  diagnostic_updater::Updater updater_{this};
-
-  const std::string input_twist_topic_type_;
-
-  /** \brief Output TF frame the concatenated points should be transformed to. */
-  std::string output_frame_;
-
-  /** \brief Input point cloud topics. */
-  // XmlRpc::XmlRpcValue input_topics_;
-  std::vector<std::string> input_topics_;
-
-  /** \brief TF listener object. */
-  std::shared_ptr<tf2_ros::Buffer> tf2_buffer_;
-  std::shared_ptr<tf2_ros::TransformListener> tf2_listener_;
-
-  std::deque<geometry_msgs::msg::TwistStamped::ConstSharedPtr> twist_ptr_queue_;
-
-  std::map<std::string, sensor_msgs::msg::PointCloud2::ConstSharedPtr> cloud_stdmap_;
-  std::map<std::string, sensor_msgs::msg::PointCloud2::ConstSharedPtr> cloud_stdmap_tmp_;
+  struct Parameters
+  {
+    int maximum_queue_size;
+    double timeout_sec;
+    bool is_motion_compensated;
+    bool publish_synchronized_pointcloud;
+    bool keep_input_frame_in_synchronized_pointcloud;
+    bool publish_previous_but_late_pointcloud;
+    std::string synchronized_pointcloud_postfix;
+    std::string input_twist_topic_type;
+    std::vector<std::string> input_topics;
+    std::string output_frame;
+    std::vector<double> lidar_timestamp_offsets;
+    std::vector<double> lidar_timestamp_noise_window;
+  } params_;
+
+  double current_concatenate_cloud_timestamp_{0.0};
+  double lastest_concatenate_cloud_timestamp_{0.0};
+  bool drop_previous_but_late_pointcloud_{false};
+  bool publish_pointcloud_{false};
+  double diagnostic_reference_timestamp_min_{0.0};
+  double diagnostic_reference_timestamp_max_{0.0};
+  std::unordered_map<std::string, double> diagnostic_topic_to_original_stamp_map_;
+
+  std::shared_ptr<CombineCloudHandler> combine_cloud_handler_;
+  std::shared_ptr<CloudCollector> cloud_collector_;
+  std::list<std::shared_ptr<CloudCollector>> cloud_collectors_;
   std::mutex mutex_;
+  std::unordered_map<std::string, double> topic_to_offset_map_;
+  std::unordered_map<std::string, double> topic_to_noise_window_map_;
+
+  // subscribers
+  std::vector<rclcpp::Subscription<sensor_msgs::msg::PointCloud2>::SharedPtr> pointcloud_subs;
+  rclcpp::Subscription<geometry_msgs::msg::TwistWithCovarianceStamped>::SharedPtr twist_sub_;
+  rclcpp::Subscription<nav_msgs::msg::Odometry>::SharedPtr odom_sub_;
+
+  // publishers
+  rclcpp::Publisher<sensor_msgs::msg::PointCloud2>::SharedPtr concatenate_cloud_publisher_;
```

Comment:
Spelling:
```suggestion
  rclcpp::Publisher<sensor_msgs::msg::PointCloud2>::SharedPtr concatenated_cloud_publisher_;
```

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/8300/commits/07cb7534ea90d6d3e32948c32eb57d87c7bf8722